src/dmd/dtoh.d

```diff
@@ -3069,6 +3078,10 @@ const(char*) keywordClass(const Identifier ident)
             return null;
 
         default:
+            // Identifiers starting with __ are reserved
+            if (name.length >= 2 && name[0..2] == "__")
+                return "reserved identifier in C++";
```

Comment:
I get why this is here, but if this is to replace the manually updated headers then some of the identifiers this removes from the frontend \* are used in LDC in [asm-x86.h](https://github.com/ldc-developers/ldc/blob/58ee56a2100eb301c55220933dd2a84d3a7d2843/gen/asm-x86.h#L2146) and others may be used elsewhere (I haven't done an exhaustive search)

Reply:
cc @kinke

Reply:
None of the static Id variables are exposed to C++ since idgen was removed. Just use `Identifier.idPool` to get the pointer, or define the static variables locally on the codegen/glue layer side if having them really matter.

Reply:
For reference, LDC actually uses a [modified revision of `id.h`](https://github.com/ldc-developers/ldc/blob/58ee56a2100eb301c55220933dd2a84d3a7d2843/dmd/id.h) that declares those variables.

---

The current check is simpler/ more complete but I wouldn't mind an explicit list of keywords - as long as it recognizes any declaration that will be rejected by gcc, clang, ...

Reply:
The problem with an explicit list is that there will always be more keywords to add (I didn't see any traits-like keywords `__direct_bases`, `__is_pod`, `__has_unique_object_representations`, etc...)

Reply:
Fair enough, an interal list will never be enough.
I guess a future extension could add a way to explicitly allow / disallow identifiers to deal with such scenarios.